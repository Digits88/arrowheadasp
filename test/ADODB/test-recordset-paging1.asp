<!--#include file="../config/ADODB.inc"-->
Here is the test of the RecordSet paging features.<br>
<%
    Dim RS

    Set RS = Server.CreateObject("ADODB.RecordSet")
    RS.CursorLocation = 3
    RS.CursorType = 3
    RS.PageSize = 5
    RS.CacheSize = 5
    RS.Open "SELECT StringField, IntField FROM Test ORDER BY IntField", _
        Connect_FullDSN

    %>AbsolutePosition: <%=RS.AbsolutePosition%><br />
    AbsolutePage: <%=RS.AbsolutePage%><br />
    PageCount: <%=RS.PageCount%><br />
<%
    While not RS.Eof
        %>Value: <%=RS("StringField")%> / <%=RS("IntField")%><br>
        AbsolutePosition: <%=RS.AbsolutePosition%><br />
        AbsolutePage: <%=RS.AbsolutePage%><br />
        PageCount: <%=RS.PageCount%><br />
<%
        RS.MoveNext
    Wend

    %>AbsolutePosition: <%=RS.AbsolutePosition%><br />
    AbsolutePage: <%=RS.AbsolutePage%><br />
    PageCount: <%=RS.PageCount%><br />
<%

    RS.Close
%>

The End.
