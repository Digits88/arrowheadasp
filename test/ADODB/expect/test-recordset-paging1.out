
Here is the test of the RecordSet paging features.<br>
AbsolutePosition: 1<br />
    AbsolutePage: 1<br />
    PageCount: 3<br />
Value: First Row / 1<br>
        AbsolutePosition: 1<br />
        AbsolutePage: 1<br />
        PageCount: 3<br />
Value: Second Row / 2<br>
        AbsolutePosition: 2<br />
        AbsolutePage: 1<br />
        PageCount: 3<br />
Value: Third Row / 3<br>
        AbsolutePosition: 3<br />
        AbsolutePage: 1<br />
        PageCount: 3<br />
Value: Fourth Row / 4<br>
        AbsolutePosition: 4<br />
        AbsolutePage: 1<br />
        PageCount: 3<br />
Value: Fifth Row / 5<br>
        AbsolutePosition: 5<br />
        AbsolutePage: 1<br />
        PageCount: 3<br />
Value: Sixth Row / 6<br>
        AbsolutePosition: 6<br />
        AbsolutePage: 2<br />
        PageCount: 3<br />
Value: Seventh Row / 7<br>
        AbsolutePosition: 7<br />
        AbsolutePage: 2<br />
        PageCount: 3<br />
Value: Eigth Row / 8<br>
        AbsolutePosition: 8<br />
        AbsolutePage: 2<br />
        PageCount: 3<br />
Value: Ninth Row / 9<br>
        AbsolutePosition: 9<br />
        AbsolutePage: 2<br />
        PageCount: 3<br />
Value: Tenth Row / 10<br>
        AbsolutePosition: 10<br />
        AbsolutePage: 2<br />
        PageCount: 3<br />
Value: Eleventh Row / 11<br>
        AbsolutePosition: 11<br />
        AbsolutePage: 3<br />
        PageCount: 3<br />
Value: Twelfth Row / 12<br>
        AbsolutePosition: 12<br />
        AbsolutePage: 3<br />
        PageCount: 3<br />
Value: Thirteenth Row / 13<br>
        AbsolutePosition: 13<br />
        AbsolutePage: 3<br />
        PageCount: 3<br />
Value: Fourteenth Row / 14<br>
        AbsolutePosition: 14<br />
        AbsolutePage: 3<br />
        PageCount: 3<br />
Value: Fifteenth Row / 15<br>
        AbsolutePosition: 15<br />
        AbsolutePage: 3<br />
        PageCount: 3<br />
AbsolutePosition: -^.+$<br />
    AbsolutePage: -^.+$<br />
    PageCount: 3<br />


The End.
